### PR TITLE
Feature/certificate rules

### PR DESCRIPTION
**Pull request by alexmoroianu**

### CODE REVIEW

File: services/distribution/src/test/resources/dgc/country-list.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+[
```
Comment (roesslerj) [inline services/distribution/src/test/resources/dgc/country-list.json:1]: Why is this list added twice (the other time at services/distribution/src/main/resources/dgc/country-list.json)?

File: services/distribution/src/test/resources/dgc/dcc-validation-rule.json [hunk 2]
```diff
@@ -0,0 +1,146 @@
+{
```
Comment (roesslerj) [inline services/distribution/src/test/resources/dgc/dcc-validation-rule.json:1]: Same: Seems to be a doublette of the file at services/distribution/src/main/resources/dgc/

**Review by roesslerj — DISMISSED** [review]

File: services/distribution/src/test/resources/dgc/dcc-validation-rule.json (hunk 2 above)
Comment (roesslerj) [reply services/distribution/src/test/resources/dgc/dcc-validation-rule.json:1]: And the files below also...?

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProvider.java [hunk 3]
```diff
@@ -61,6 +73,42 @@ private DirectoryOnDisk constructArchiveToPublish(DigitalGreenCertificate dgcCon
       logger.info("Writing digital green certificate to {}/{}/{}.", dgcDirectory.getName(), languageDirectory.getName(),
           archiveToPublish.getName());
     }
+
+    dgcDirectory.addWritable(getOnboardedCountriesArchive());
+    dgcDirectory.addWritable(getRulesArchive(RuleType.Acceptance, ACCEPTANCE_RULES));
+    dgcDirectory.addWritable(getRulesArchive(RuleType.Invalidation, INVALIDATION_RULES));
```
Comment (hilmarf) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProvider.java:79]: kind of redundant information.... 
`RuleType.Acceptance, ACCEPTANCE_RULES`
`RuleType.Invalidation, INVALIDATION_RULES` 
why isn't one or the other sufficient?

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/Application.java [hunk 4]
```diff
@@ -30,9 +31,10 @@
 @SpringBootApplication
 @EnableJdbcRepositories(basePackages = "app.coronawarn.server.common.persistence")
 @EntityScan(basePackages = "app.coronawarn.server.common.persistence")
-@ComponentScan({"app.coronawarn.server.common.persistence", "app.coronawarn.server.services.distribution",
```
Comment (ioangut) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/Application.java:33]: diifferent spacing settings

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProvider.java (hunk 3 above)
Comment (alexmoroianu) [reply services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProvider.java:79]: One is the name of the archive which from the techspec should be "acceptance-rules" or "invalidation-rules" and the other is the enum for which the filtering of rules is made. I could let only the enum but then I have to treat the archive name condition in the method..

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/client/CloudDccFeignHttpClientProvider.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  /**
+   * Creates a FeignClient.
+   */
+  @Override
+  @Bean
+  public Client createFeignClient() {
+    return new ApacheHttpClient(
+        federationHttpClientFactory(connectionPoolSize).createBuilder().build());
```
Comment (hilmarf) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/client/CloudDccFeignHttpClientProvider.java:55]: @AliceBilc - if you're reading into the feign client stuff... when are the opened sockets actually released?

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/client/ProdDigitalCovidCertificateClient.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  private DigitalCovidCertificateFeignClient digitalCovidCertificateClient;
+
+  public ProdDigitalCovidCertificateClient(DigitalCovidCertificateFeignClient digitalCovidCertificateFeignClient) {
+    this.digitalCovidCertificateClient = digitalCovidCertificateFeignClient;
+  }
+
+  @Override
+  public List<String> getCountryList() {
+    logger.info("Get country list");
```
Comment (hilmarf) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/client/ProdDigitalCovidCertificateClient.java:29]:
```suggestion
    logger.debug("Get country list");
```

File: services/distribution/src/main/resources/dgc/rule_1.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
+{
+  "Identifier": "RR-NL-0000",
```
Comment (hilmarf) [inline services/distribution/src/main/resources/dgc/rule_1.json:2]: move to `src/**test**/resources`

File: services/distribution/src/main/resources/dgc/rule_3.json [hunk 8]
```diff
@@ -0,0 +1,53 @@
+{
+  "Identifier": "RR-NL-0003",
```
Comment (hilmarf) [inline services/distribution/src/main/resources/dgc/rule_3.json:2]: move to **test**:  `src/test/resources` ???

File: services/distribution/src/main/resources/dgc/rules.json [hunk 9]
```diff
@@ -0,0 +1,20 @@
+[
+  {
```
Comment (hilmarf) [inline services/distribution/src/main/resources/dgc/rules.json:2]: move to **test**:  `src/test/resources` ???

File: services/distribution/src/main/resources/dgc/valuesets.json [hunk 10]
```diff
@@ -0,0 +1,34 @@
+[
+  {
+    "id": "country-2-codes",
```
Comment (hilmarf) [inline services/distribution/src/main/resources/dgc/valuesets.json:3]: move to **test**:  `src/test/resources` ???

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProviderTest.java [hunk 11]
```diff
@@ -1,7 +1,9 @@
 package app.coronawarn.server.services.distribution.assembly.component;
 
-import static org.junit.Assert.assertTrue;
+
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (hilmarf) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProviderTest.java:6]: 👍

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/dgc/DigitalCovidCertificateClientTest.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+  public static final String DE_HASH = "6821d518570fe9f4417c482ff0d2582a7b6440f243a9034f812e0d71611b611f";
+  public static final String DE = "DE";
+
+  @Autowired
+  private DigitalCovidCertificateClient digitalCovidCertificateClient;
+
+  @Test
+  public void testCountryList() throws DigitalCovidCertificateException {
+    System.out.println(digitalCovidCertificateClient.getCountryList());
```
Comment (hilmarf) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/dgc/DigitalCovidCertificateClientTest.java:32]: 👎

**Review by hilmarf — APPROVED** [review]

**Review by roesslerj — APPROVED** [review]

**Pull request merged**